server/util/scratchspace/scratchspace.go

```diff
@@ -0,0 +1,54 @@
+package scratchspace
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+)
+
+var (
+	tempDir_doNotUseDirectly = flag.String("storage.tempdir", filepath.Join(os.TempDir(), "buildbuddy-scratch"), "Staging directory for ephemeral files. If unspecified, defaults to 'TEMPDIR/buildbuddy' where TEMPDIR is the OS-specific temp dir.")
```

Comment:
I think it would be more straightforward to have your flag default to os.TempDir() and then your "TempDir()" func below can *always* append "buildbuddy-scratch" (which could could make a const?). Then it becomes impossible for someone to accidentally blow away the contents of some dir if they accidentally set --storage.tempdir to "/home/tylerw" or something.